### PR TITLE
[Aptos Framework] add `set_validator_num_blocks` to the `Stake.move`

### PR DESCRIPTION
**Pull request by mkurnikov**
### Description

It's needed for any tests that touch validators, otherwise, they fail with div-by-zero. 

### Test Plan

It's used in the `end_epoch`, no need for a separate tests. 

@movekevin

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2126)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/configs/stake.move [hunk 1]
```diff
@@ -1606,10 +1606,15 @@ module aptos_framework::stake {
         assert_validator_state(signer::address_of(account), 100, 0, 0, 0, 0);
     }
 
+    #[test_only]
+    public fun set_validator_perf_num_blocks(num_blocks: u64) acquires ValidatorPerformance {
+        borrow_global_mut<ValidatorPerformance>(@aptos_framework).num_blocks = num_blocks;
+    }
+
     #[test_only]
     fun end_epoch() acquires StakePool, StakePoolEvents, TestCoinCapabilities, ValidatorConfig, ValidatorPerformance, ValidatorSet, ValidatorSetConfiguration {
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/configs/stake.move:1615]: Why not make this public instead? Do you have use cases for setting a custom num_blocks value in a test?

Comment (mkurnikov) [reply aptos-move/framework/aptos-framework/sources/configs/stake.move:1615]: Ok, I tried, but it doesn't work. 
I have a code like this for epoch changes in tests
```
    public entry fun epoch(aptos_framework: &signer) {
        timestamp::update_global_time_for_test(timestamp::now_microseconds() + 1);
        stake::set_validator_num_blocks(1);
        reconfiguration::force_reconfigure(aptos_framework);
    }
```
If I change the second line into `end_epoch()`, it's going to run `stake::on_new_epoch()` twice and fail with div-by-zero. 
If I remove the second line, `reconfiguration::reconfigure_()` is not called, and the state isn't properly updated.

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/configs/stake.move:1615]: Oh I see that makes sense if you force reconfigure.

**Review by movekevin — APPROVED** [review]

**Comment by mkurnikov** [comment]
Any chance to see it in `devnet` soon?

**Comment by movekevin** [comment]
Adding skip forge label only because this is test only changes. Definitely would not want to skip forge at all for any non-test code

**Pull request merged**